### PR TITLE
Per @jimfulton, handle_all_serials shouldn't be sniffing in ZODB5

### PR DESCRIPTION
**Pull request by jamadden**
So try that. This slightly simplifies _dostore.

There are lots of other callers still so I didn't remove the functions entirely (but I could try if desired).

### CODE REVIEW

**Comment by jamadden** [comment]
Turns out that at least the tests pass sequences of non-bytes to this method on Python 3 (probably str?), so I had to remove that assert.

File: src/ZODB/tests/StorageTestBase.py [hunk 1]
```diff
@@ -194,10 +178,7 @@ def _dostore(self, oid=None, revid=None, data=None,
             r1 = self._storage.store(oid, revid, data, '', t)
```
Comment (jimfulton) [inline src/ZODB/tests/StorageTestBase.py]: We don't use r1 and r2, so why assign them?

**Comment by jimfulton** [comment]
Gutting the methods is confusing.  Let's not leave this for a future code reader to stumble on. (It will probably be me and I just hurt myself for the last few minutes on this :) ).

We should remove these functions and their call points, or just remove the calls in dostore, if that's what was hurting RelStorage.

File: src/ZODB/tests/StorageTestBase.py (hunk 1 above)
Comment (jamadden) [inline src/ZODB/tests/StorageTestBase.py]: I was trying to make the diff as minimal as possible. I can clean that up.

**Comment by jamadden** [comment]
> On Jul 12, 2016, at 11:00, Jim Fulton <notifications@github.com> wrote:
> 
> Gutting the methods is confusing. Let's not leave this for a future code reader to stumble on. (It will probably be me and I just hurt myself for the last few minutes on this :) ).
> 
> We should remove these functions and their call points, or just remove the calls in dostore, if that's what was hurting RelStorage.

OK, I can remove the functions and their callers. That will involve touching three more files, but I think it's mostly copy-pasted code.

**Comment by jamadden** [comment]
Travis appears to be green with the removal. I say appears because github has been lagging a lot lately.

**Pull request merged**